File: 2016_03_20_08_xp/keil/handle_v4/USER/code/script/findpath.py
# -*- coding: utf-8 -*-
# @Author: obsidian
# @Date:   2016-02-28 08:38:07
# @Last Modified by:   obsidian
# @Last Modified time: 2016-02-28 09:40:17
from ana import arr
for i in range(20):
    for k in range(80):
        pass
point = []
def getdis(i,k):
    for di in range(31):
        if i + di >= 80:
            return 0
        if arr[i + di - 15][k][0] == 31:
            print i + di - 15,k
            arr[i + di - 15][k].append(abs(di - 15))
            return abs(di - 15)
def getneibor(i,k):
    global arr
    arr[i][k][1] = 1
    point.append([i,k,getdis(i,k)])

    if i < 20 or i > 50:
        return
    if k < 10 or k > 60:
        return
    for ix in range(9):
        for iy in range(9):
            #print arr[ix + i - 5][iy + k - 5][0]
            if arr[ix + i - 5][iy + k - 5][0] == 15 and arr[ix + i - 5][iy + k - 5][1] == 0:
                getneibor(ix + i - 5,iy + k - 5)
                
getneibor(40,40)
# print arr
print point

File: 2016_03_20_08_xp/keil/handle_v4/USER/code/script/findpath.c
/*
* @Author: obsidian
* @Date:   2016-02-28 09:43:27
* @Last Modified by:   obsidian
* @Last Modified time: 2016-02-28 14:28:29
*/

#include "arr.h"
#include "math.h"
signed char record[80][60][2] = {0};
signed short point[100][3] = {0};
unsigned char is = 0;
void get_dis(i,k)
{
    signed char di = 0;
    for(di = -15; di < 15;di ++)
    {
        if(i + di >= 80){return;}
        if(arr[i + di][k] == 31)
        {
            record[i][k][1] = di;
        }
    }
}

void getneighbor(unsigned char i,unsigned char k)
{
    signed char ix = 0;
    signed char iy = 0;
    record[i][k][0] = 1;
    if(i < 20 || i > 50 || k < 10 || k > 60)
    {return;}
    
    get_dis(i,k);
    point[is][0] = i;
    point[is][1] = k;
    point[is][2] = record[i][k][1];
    is ++;
    for(ix = -4; ix <= 4; ix ++)
    {
        for(iy = -4 ; iy <= 4; iy ++)
        {
            if(arr[ix + i][iy + k] == 15 && record[ix + i][iy + k][0] == 0)
            {
                getneighbor(ix + i, iy + k);
            }
        }
    }
}
float ave()
{
}
int main() {
    getneighbor(40,40);
    return 0;
}
